src/iuse.cpp

```diff
@@ -2010,75 +2010,83 @@ int iuse::sew_advanced(player *p, item *it, bool, const tripoint& )
         return mcopy;
     };
 
-    // TODO: Wrap all the mods into structs, maybe even json-able
+    // @todo: Wrap all the mods into structs, maybe even json-able
     // All possible mods here
-    std::array<std::string, 4> clothing_mods{
+    const std::array<const std::string, 4> clothing_mods{
         { "wooled", "furred", "leather_padded", "kevlar_padded" }
     };
 
     // Materials those mods use
-    std::array<std::string, 4> mod_materials{
+    const std::array<const std::string, 4> mod_materials{
         { "felt_patch", "fur", "leather", "kevlar_plate" }
     };
 
+    const inventory &crafting_inv = p->crafting_inventory();
+    const auto &req = requirement_id("filament").obj();
+
     // Cache available materials
-    std::map< itype_id, bool > has_enough;
+    std::map< itype_id, bool > has_enough_materials;
     const int items_needed = mod->volume() / 750_ml + 1;
-    const inventory &crafting_inv = p->crafting_inventory();
     // Go through all discovered repair items and see if we have any of them available
     for( auto &material : mod_materials ) {
-        has_enough[material] = crafting_inv.has_amount( material, items_needed );
+        has_enough_materials[material] = crafting_inv.has_amount( material, items_needed );
     }
 
+    const int filament_needed = mod->volume() / 125_ml + 10;     // We need extra filament to lose it on bad rolls
+    int batch_size = filament_needed;
```

Comment:
Instead of using `batch_size`, you can multiply a requirement:
```
const auto req = requirement_id("filament").obj() * filament_needed;
```

It would be safer than using `batch_size`, because someday batch rules may change to allow lower costs or similar effects.